internal/ui/ime_test.go

```
@@ -0,0 +1,263 @@
+	anchor.set(14, 9, true)
+	w := &cursorOutputWriter{out: &out, anchor: anchor}
+
+	if _, err := w.Write([]byte("shell")); err != nil {
+		t.Fatal(err)
+	}
+	if got := out.String(); got != "shell" {
+		t.Fatalf("ordinary output changed to %q", got)
+	}
+
+	out.Reset()
+	if _, err := w.Write([]byte(ansi.SetAltScreenSaveCursorMode + "frame")); err != nil {
```

golangci-lint (2.12.2)

[error] SA1019: ansi.SetAltScreenSaveCursorMode is deprecated: use [ModeAltScreenSaveCursor] instead. 

(staticcheck)